File: languages/php/src/BitwardenSDK.php
<?php

namespace Bitwarden\Sdk;

use Bitwarden\Sdk\Schemas\AccessTokenLoginRequest;
use Bitwarden\Sdk\schemas\ClientSettings;
use Bitwarden\Sdk\Schemas\Command;
use FFI;
use Swaggest\JsonDiff\Exception;


class BitwardenSDK
{
    private BitwardenLib $bitwarden_lib;

    private ClientSettings $clientSettings;

    public ProjectsClient $projectsClient;

    public SecretsClient $secretsClient;

    private CommandRunner $commandRunner;

    private FFI\CData $handle;

    public function __construct()
    {
        $this->clientSettings = new ClientSettings();
        $this->clientSettings->apiUrl = getenv('API_URL') ?: 'https://api.bitwarden.com';
        $this->clientSettings->identityUrl = getenv('IDENTITY_URL') ?: 'https://identity.bitwarden.com';
        $this->clientSettings->userAgent = getenv('USER_AGENT') ?: 'SDK';
        $this->clientSettings->deviceType = getenv('DEVICE_TYPE') ?: 'SDK';

        $this->bitwarden_lib = new BitwardenLib();
        $this->handle = $this->bitwarden_lib->init($this->clientSettings);

        $this->commandRunner = new CommandRunner($this->bitwarden_lib, $this->handle);
        $this->projectsClient = new ProjectsClient($this->commandRunner);
        $this->secretsClient = new SecretsClient($this->commandRunner);
    }

    /**
     * @throws \Exception
     */
    public function authorize(string $access_token)
    {
        $access_token_request = new AccessTokenLoginRequest();
        $access_token_request->accessToken = $access_token;
        $command = new Command();
        $command->accessTokenLogin = $access_token_request->jsonSerialize();
        $result = $this->commandRunner->run($command);
        if (!isset($result->authenticated)) {
            throw new \Exception("Authorization error");
        }

        if ($result->authenticated == False) {
            throw new \Exception("Unauthorized");
        }

        return $result;
    }

    public function free_mem()
    {
        $this->bitwarden_lib->free_mem();
    }
}

File: languages/ruby/bitwarden_sdk/lib/bitwarden-sdk.rb
# frozen_string_literal: true

require 'json'
require 'dry-types'

require_relative 'schemas'
require_relative 'extended_schemas/schemas'
require_relative 'command_runner'
require_relative 'bitwarden_lib'
require_relative 'bitwarden_error'
require_relative 'projects'
require_relative 'secrets'

module BitwardenSDK
  class BitwardenSettings
    attr_accessor :api_url, :identity_url

    def initialize(api_url, identity_url)
      @api_url = api_url
      @identity_url = identity_url
    end
  end

  class BitwardenClient
    attr_reader :bitwarden, :project_client, :secrets_client

    def initialize(bitwarden_settings)
      client_settings = ClientSettings.new(
        api_url: bitwarden_settings.api_url,
        identity_url: bitwarden_settings.identity_url,
        user_agent: 'Bitwarden RUBY-SDK',
        device_type: nil
      )

      @bitwarden = BitwardenLib
      @handle = @bitwarden.init(client_settings.to_json)
      @command_runner = CommandRunner.new(@bitwarden, @handle)
      @project_client = ProjectsClient.new(@command_runner)
      @secrets_client = SecretsClient.new(@command_runner)
    end

    def authorize(access_token_login)
      access_token_request = AccessTokenLoginRequest.new(access_token: access_token_login)
      @command_runner.run(SelectiveCommand.new(access_token_login: access_token_request))
    end

    def free_mem
      @bitwarden.free_mem(@handle)
    end
  end
end
